lib/elixir/lib/enum.ex

```diff
@@ -932,7 +932,8 @@ defmodule Enum do
 
   @doc """
   Returns a new enumerable appending the result of invoking `fun` on
-  each corresponding item of `enumerable`.
+  each corresponding item of `enumerable` and flattening
+  the given results (only one level deep)..
```

Comment:
I'm not sure we want this change because the documentation says that we **append** the result of invoking `fun`, which should be explanatory. Also the examples show this behaviour pretty clearly I think. Wdyt? I'm not convinced we don't need to be more specific yet, so I'm open to discussion :)